Sources/PubNub/Helpers/Crypto/Header/CryptorHeaderWithinStreamFinder.swift

```diff
@@ -30,33 +30,37 @@ import Foundation
 struct CryptorHeaderWithinStreamFinder {
   let stream: InputStream
   
-  // Attempts to find CryptorHeader at the beginning of the given InputStream.
-  // Returns CryptorHeader (if any) and InputStream starting with encoded content
-  func findHeader() throws -> (header: CryptorHeader, continuationStream: InputStream) {
-    let possibleHeaderBytes = read(maxLength: 100)
-    let parsingRes = try CryptorHeaderParser(data: possibleHeaderBytes).parseAndReturnProcessedBytes()
-    let noOfBytesProcessedByParser = parsingRes.bytesProcessed.count
+  // Attempts to find CryptorHeader in the given InputStream.
+  // Returns InputStream that immediately follows CryptorHeader
+  func findHeader() throws -> (header: CryptorHeader, cryptorDefinedData: Data, continuationStream: InputStream) {
+    let buffer = read(maxLength: 1024)
+    let header = try CryptorHeaderParser(data: buffer).parse()
+    let headerLength = header.length()
+    
     let continuationStream: InputStream
+    let cryptorDefinedData: Data
     
-    switch parsingRes.header {
+    switch header {
     case .none:
-      continuationStream = MultipartInputStream(
-        inputStreams: [
-          InputStream(data: possibleHeaderBytes),
-          stream
-        ]
-      )
-    default:
-      continuationStream = MultipartInputStream(
-        inputStreams: [
-          InputStream(data: possibleHeaderBytes.suffix(from: noOfBytesProcessedByParser)),
-          stream
-        ]
-      )
+      // There is no CryptorHeader, so all supplied buffer bytes belong to the contents of the File
+      cryptorDefinedData = Data()
+      continuationStream = MultipartInputStream(inputStreams: [InputStream(data: buffer), stream])
+    case .v1(_, let dataLength):
+      // Detects whether it's safe to extract metadata
+      guard headerLength + dataLength < buffer.count else {
+        throw PubNubError(.decryptionFailure, additional: ["Cannot extract metadata for CryptorHeader v1"])
+      }
+      // Extracts Cryptor-defined data from the supplied buffer
+      cryptorDefinedData = buffer.subdata(in: headerLength..<headerLength + dataLength)
```

Comment:
Just a note, that there can be no cryptor-defined data at all or custom implementation will handle it in a different way.

Reply:
I think I will rework that when implementing another Cryptor. Fortunately, that's the internal implementation so it won't break customers in terms of any API. The question is where do you extract `metadata` for `V1` header? Is it inside `CryptorModule` or somewhere else?

Reply:
Since `CryptoModule` after data pre-processing (header extraction) calls to actual cryptor implementations which expect specific types (_encrypt_ expects `Data` and _decrypt_ expects `EncrpyptedData` which contains cryptor-defined portion of data and encrypted data). It is module who extracts `metadata` and puts it into `EncrpyptedData`.